### ISSUE TITLE
Port ServiceAuthenticationBehavior class

### ISSUE THREAD

**Issue by mconnew** [CoreWCF/CoreWCF#92]
The original code can be found [here](https://github.com/microsoft/referencesource/blob/master/System.ServiceModel/System/ServiceModel/Description/ServiceAuthenticationBehavior.cs).  
This will also need AuthenticationBehavior ported (can be found [here](https://github.com/microsoft/referencesource/blob/master/System.ServiceModel/System/ServiceModel/Dispatcher/AuthenticationBehavior.cs)) and the relevant call added to the dispatcher.

**Comment by dgxhubbard** [comment]
I am confused. There is already a ServiceAuthorizationBehavior class in CoreWCF/Descriptions and a comparison of [here](https://github.com/microsoft/referencesource/blob/master/System.ServiceModel/System/ServiceModel/Description/ServiceAuthenticationBehavior.cs) and class in CoreWCF show role provider related code is removed. This code was added by @mconnew. So is a port needed?

**Comment by mconnew** [comment]
You're right. I swear I did do a search before I opened the issue. I must have been searching against an older commit. Github has a habit of picking random older commits when doing a text search so I think I might have used a text search as a starting point for a file search (press t when in the code view and you can search for file names).  
We don't have any tests for it though. Would you be willing to write the tests? If so I can give you a quick overview of what the tests need to do if you aren't sufficiently familiar with this api.

**Comment by dgxhubbard** [comment]
Yes I can. Any background you provide will be appreciated.
